src/main/java/org/candlepin/util/ContentOverrideValidator.java

```diff
@@ -98,7 +98,7 @@ public void validate(Collection<? extends ContentOverrideDTO> overrides) {
                         invalidProps.add(name != null ? name : "null");
                     }
 
-                    if (value == null || value.length() == 0 || value.length() > MAX_VALUE_LENGTH) {
+                    if (value == null || value.length() == 0) {
```

Comment:
Even if we have a database column that has no explicit limit, we should not be accepting unlimited length strings. We should decide on a long, but still reasonable limit here regardless what we do with the column type.

My suggestion here would be 1024 characters. Maaaaaybe 2048 if we've got some customers out there that want to push that field to its extremes. This also has the knock on effect of guiding where the database type goes.

Reply:
I think that 1024 characters is a reasonable amount, and we can always easily increase it if there are some customers that need it increased.

Reply:
FYI: the case in point [1] has a length of 950 characters, so 1024 would be still too close (but higher, yes) to that limit; hence, I'd suggest to use at least 2048 or even 5000 characters (not sure whether there is a need for sticking to powers of 2).

[1] https://bugzilla.redhat.com/show_bug.cgi?id=1969341

Reply:
I updated the character limit to 2048

Reply:
> not sure whether there is a need for sticking to powers of 2

Nope, just faster math in some cases and nicer bitwise ops -- both things which don't matter here. 

It's important to note that as this column gets larger and larger, we're up against some real painful facts about it. Namely, as of today, we have some 226 million rows to worry about, and querying it is already incredibly painful. While it would be nice to use a text type here and not need to worry about updating it in the future, query performance would likely suffer. But, we can't just slap a huge number on a varchar type for the same reason. If we exceed the maximum row size, we're going to be up against the same perf problem TEXT has, as, at least in the case of MySQL/MariaDB, variable length columns will get chopped up and stored off-table.

That said, given what I'm seeing in the table right now, I would argue that we should not exceed ~4096 characters. It's well above the use case, and the sheer storage requirements for such absurd potential table sizes with such a limit is likely going to get us a talking to from our friends in IT.